pkg/compose/convergence.go

```diff
@@ -123,11 +147,11 @@ func (c *convergence) ensureService(ctx context.Context, project *types.Project,
 
 	sort.Slice(containers, func(i, j int) bool {
 		// select obsolete containers first, so they get removed as we scale down
-		if obsolete, _ := mustRecreate(service, containers[i], recreate); obsolete {
+		if obsolete, _ := mustRecreate(project, service, containers[i], recreate, volumes); obsolete {
```

Comment:
you don't need to check volumes here: if any was not matching config, it has just been recreated in `ensureVolumes` and service was removed, so `ensureService` will recreate service from scratch

Reply:
If you go back to the first issue, changing the name of the volume won't update the mount in the container. This piece of code is necessary for it

Reply:
you don't get my point: As you detect volume must be recreated within `ensureVolume`, you remove all containers with a mount it before you can delete and recreate volume. Later when you enter `ensureService`, as those containers have been removed, this code will never run and is useless.

Reply:
Applied changes, still need to account for the external volume name change